metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -528,10 +531,70 @@ def _populate_view_downstream_lineage(self, conn: SnowflakeConnection) -> None:
             f"Found {self.report.num_view_to_table_edges_scanned} View->Table edges."
         )
 
-    def warn(self, key: str, reason: str) -> None:
-        self.report.report_warning(key, reason)
-        self.logger.warning(f"{key} => {reason}")
+    def update_upstream_tables_lineage(self, upstream_tables, lineage):
+        for lineage_entry in sorted(
+            lineage.upstreamTables.values(), key=lambda x: x.upstreamDataset
+        ):
+            upstream_table_name = lineage_entry.upstreamDataset
+            upstream_table_urn = builder.make_dataset_urn_with_platform_instance(
+                self.platform,
+                upstream_table_name,
+                self.config.platform_instance,
+                self.config.env,
+            )
+            upstream_table = UpstreamClass(
+                dataset=upstream_table_urn,
+                type=DatasetLineageTypeClass.TRANSFORMED,
+            )
+            upstream_tables.append(upstream_table)
+
+    def update_upstream_columns_lineage(
+        self, dataset_urn, finegrained_lineages, lineage
+    ):
+        for col, col_upstreams in lineage.columnLineages.items():
+            for fine_upstream in col_upstreams.upstreams:
```

Comment:
No nested for loops any more. Can we extract into a separate method.

Reply:
refractored.